File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.java
import java.util.*;

class Solution {
    public int[] solution(int l, int r) {
        List<Integer> result = new ArrayList<>();
        Queue<Integer> q = new LinkedList<>();
        

        q.add(5);
        
        while (!q.isEmpty()) {
            int num = q.poll();
            
            if (num > r) continue;
            
            if (num >= l) {
                result.add(num);
            }
            

            if (num * 10 <= r) {
                q.add(num * 10);
                q.add(num * 10 + 5);
            }
        }
        
        if (result.isEmpty()) {
            return new int[]{-1};
        }
        

        Collections.sort(result);

        return result.stream().mapToInt(i -> i).toArray();
    }
}

File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.js
function solution(l, r) {
    const result = [];

    for(let i = 1; ; i++) {
        const num = parseInt(i.toString(2)) * 5;
        
        if(num > r) break;
        if(num >= l) result.push(num);
    }
    
    return result.length === 0 ? [-1] : result;
}